icu4c/source/i18n/number_mapper.h

```diff
@@ -196,10 +198,18 @@ struct DecimalFormatFields : public UMemory {
     LocalizedNumberFormatter formatter;
 
     /** The lazy-computed parser for .parse() */
+#if U_HAVE_ATOMICS
     std::atomic<::icu::numparse::impl::NumberParserImpl*> atomicParser = {};
+#else
+    ::icu::numparse::impl::NumberParserImpl* atomicParser = nullptr;
```

Comment:
nullptr vs {} for initialization. The corresponding currency parser initialization, six lines down, is using {}. Should be consistent.